sound/soc/sof/intel/hda-codec.c

```diff
@@ -154,12 +154,15 @@ int hda_codec_probe_bus(struct snd_sof_dev *sdev,
 
 		ret = hda_codec_probe(sdev, i, hda_codec_use_common_hdmi);
 		if (ret < 0) {
-			dev_err(bus->dev, "error: codec #%d probe error, ret: %d\n",
-				i, ret);
-			return ret;
+			dev_warn(bus->dev, "codec #%d probe error, ret: %d\n",
+				 i, ret);
+			bus->codec_mask &= ~BIT(i);
```

Comment:
right, but if you do this there could be a case where the external HDAudio is detected, but the iDISP isn't and the machine driver does not support this case. we also don't have topologies for HDAudio only.

Reply:
shouldn't we also make the codec_mask here so that we only have 2 bits valid here.

Reply:
The idea here is to just prune codecs that are not responding to commands. We have a couple of bugs where the STATESTS line is up, but the actual codec is not present or is faulty. The outcome may be a codec mask which the machine drivers cannot handle, but then it's ok to fail later at machine_check. Not sure I got what you mean by "making the codec_mask", I do update bus->codec_mask here which is the thing we pass around.

Reply:
@kv2019i I feel like patch 1 and 3 are not required if you just have patch 2. You'd have adjusted the codec_mask so that the count will always be either 1 or 2 now?

Reply:
Ack, 1+3 and 2 are for difference scenarios. See longer reply below ( https://github.com/thesofproject/linux/pull/1612#discussion_r356177999 )